Shell/Options.cpp

```diff
@@ -1019,418 +1068,423 @@ void Options::init()
-	    _forwardDemodulation.tag(OptionTag::INFERENCES);
-	    _forwardDemodulation.setRandomChoices({"all","all","all","off","preordered"});
+    _unificationWithAbstraction = ChoiceOptionValue<UnificationWithAbstraction>("unification_with_abstraction","uwa",
+                                     UnificationWithAbstraction::OFF,
+                                     {"off","interpreted_only","one_side_interpreted","one_side_constant","all","ground"});
+    _unificationWithAbstraction.description=
+      "During unification, if two terms s and t fail to unify we will introduce a constraint s!=t and carry on. For example, "
+      "resolving p(1) \\/ C with ~p(a+2) would produce C \\/ 1 !=a+2. This is controlled by a check on the terms. The expected "
+      "use case is in theory reasoning. The possible values are:"
+      "- off: do not introduce a constraint\n"
+      "- interpreted_only: only if s and t have interpreted top symbols\n"
+      "- one_side_interpreted: only if one of s or t have interpreted top symbols\n"
+      "- one_side_constant: only if one of s or t is an interpreted constant (e.g. a number)\n"
+      "- all: always apply\n"
+      "- ground: only if both s and t are ground\n"
+      "See Unification with Abstraction and Theory Instantiation in Saturation-Based Reasoning for further details.";
+    _unificationWithAbstraction.tag(OptionTag::INFERENCES);
+    _lookup.insert(&_unificationWithAbstraction);
+
+    _useACeval = BoolOptionValue("use_ac_eval","uace",true);
+    _useACeval.description="Evaluate associative and commutative operators e.g. + and *.";
+    _useACeval.tag(OptionTag::INFERENCES);
+    _lookup.insert(&_useACeval);
+
+    _inequalityNormalization = BoolOptionValue("normalize_inequalities","norm_ineq",false);
+    _inequalityNormalization.description="Enable normalizing of inequalities like s < t ==> 0 < t - s.";
+    _lookup.insert(&_inequalityNormalization);
+    _inequalityNormalization.tag(OptionTag::INFERENCES);
+
+    auto choiceArithmeticSimplificationMode = [&](vstring l, vstring s, ArithmeticSimplificationMode d)
+    { return ChoiceOptionValue<ArithmeticSimplificationMode>(l,s,d, {"force", "cautious", "off", }); };
+    _cancellation = choiceArithmeticSimplificationMode(
+       "cancellation", "canc",
+       ArithmeticSimplificationMode::OFF);
+    _cancellation.description = "Enables the rule cancellation around additions as described in the paper Making Theory Reasoning Simpler ( https://easychair.org/publications/preprint/K2hb ). \
+                                In some rare cases the conclusion may be not strictly simpler than the hypothesis. With `force` we ignore these cases, violating the ordering and just simplifying \
+                                anyways. With `cautious` we will generate a new clause instead of simplifying in these cases.";
+    _lookup.insert(&_cancellation);
+    _cancellation.tag(OptionTag::INFERENCES);
+    _cancellation.setExperimental();
```

Comment:
I think we could remove these setExperimental calls as the rules are in master, and in a paper, right?

Reply:
I said these, but i actually just meant this one.